libraries/AP_TemperatureSensor/AP_TemperatureSensor_Backend.cpp

```diff
@@ -63,34 +63,41 @@ void AP_TemperatureSensor_Backend::set_temperature(const float temperature)
 
 void AP_TemperatureSensor_Backend::update_external_libraries(const float temperature)
 {
-#if HAL_WITH_ESC_TELEM
-    AP_ESC_Telem_Backend::TelemetryData t;
-#endif
-
     switch ((AP_TemperatureSensor_Params::Source)_params.source.get()) {
-#if HAL_WITH_ESC_TELEM
         case AP_TemperatureSensor_Params::Source::ESC:
+#if HAL_WITH_ESC_TELEM
```

Comment:
This is not the right way to remove the default clause; the enumeration values should be removed when the source is not available.

The problem with changing things like this is that it costs bytes for a feature even when it's not compiled in.  Removing the enumeration entry, on the other hand, has the advantage that there can't be *any* code left in which is gated on that enumeration value.

Reply:
I've removed the commit per your suggestion on discord